json-rpc/types/src/views.rs

```diff
@@ -364,6 +401,7 @@ impl From<Transaction> for TransactionDataView {
                     sequence_number: t.sequence_number(),
                     max_gas_amount: t.max_gas_amount(),
                     gas_unit_price: t.gas_unit_price(),
+                    gas_currency_code: t.gas_currency_code().to_string(),
```

Comment:
let's use gas_currency to be consistent with  "currency" field in various  script field